src/store-modules/filter-store.js

```diff
@@ -244,9 +247,12 @@ function setQuery(state) {
 }
 
 function getMediaTypeFilters(state, { mediaType, includeMature = false }) {
-  let filterKeys = [...mediaFilterKeys[mediaType]]
+  // eslint-disable-next-line no-debugger
+  debugger
+
+  let filterKeys = mediaFilterKeys[mediaType]
   if (!includeMature) {
-    filterKeys.splice(filterKeys.indexOf('mature'), 1)
+    filterKeys = filterKeys.filter((filterKey) => filterKey !== 'mature')
```

Comment:
this should explain the original bug: 

<img width="518" alt="CleanShot 2021-08-11 at 16 41 39@2x" src="https://user-images.githubusercontent.com/6351754/129100259-f3d3f3e4-de0b-4dde-96ac-ee773284df94.png">